src/java/org/apache/cassandra/db/filter/RowFilter.java

```diff
@@ -698,6 +698,7 @@ public boolean isSatisfiedBy(TableMetadata metadata, DecoratedKey partitionKey,
                 case LIKE_SUFFIX:
                 case LIKE_CONTAINS:
                 case LIKE_MATCHES:
+                case ANN:
                     {
                         assert !column.isComplex() : "Only CONTAINS and CONTAINS_KEY are supported for 'complex' types";
```

Comment:
not part of this patch but this is a poor error message, "complex" is an internal thing, can we slip in a quick improvement?

Reply:
Agreed. I've changed the `'complex'` to `collection`. Seems the simplest change and makes it much clearer to the user.